SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -525,6 +526,22 @@ void USpatialNetDriver::OnGSMQuerySuccess()
 	// If the deployment is now accepting players and we are waiting to spawn. Spawn.
 	if (bWaitingToSpawn && ClientCanSendPlayerSpawnRequests())
 	{
+		// Are we running with the same schema hash as the server?
+#if !(UE_BUILD_SHIPPING)
+		if (GetDefault<USpatialGDKSettings>()->bEnableSchemaValidationOnJoin)
+		{
+			uint32 ServerHash = GlobalStateManager->GetSchemaHash();
+			if (ClassInfoManager->SchemaDatabase->SchemaDescriptorHash != ServerHash)
+			{
+				UE_LOG(LogSpatialOSNetDriver, Warning, TEXT("Your schema hash does match the server, this may cause problems. Our hash: %lu server hash: %lu"), (unsigned long)ClassInfoManager->SchemaDatabase->SchemaDescriptorHash, (unsigned long)ServerHash);
```

Comment:
Why the casts and %lu? Just use %d?

Reply:
It was to deal with the potential of %d or %lu expecting different sizes on different platforms but looks like int is 32-bit across the board anyway - so will just use %u instead